Review article: How To Setup The Dev Env

ivan9989-pkj opened the pull request:
Review for the Torrust Webise blog tutorial: [How To Setup The Dev Env](https://torrust.com/how-to-setup-the-development-environment).

Installation has changed.

ivan9989-pkj committed:
Actualizacion de documentacion

Actualizamos a una versión mas nueva la documentación de la pagina deploying-torrust-to-production

ivan9989-pkj commented:
I recommend looking at it to see if you think it is well done, report if you see any errors so that I can correct them.

ivan9989-pkj committed:
Actualización

Hemos actualizado el archivo how-to-setup-deploying

ivan9989-pkj committed:
Actulización

Actualizamos la pagina how-to-setup-the-development-environment

josecelano commented:
> I recommend looking at it to see if you think it is well done, report if you see any errors so that I can correct them.

OK, @ivan9989-pkj, I'm reviewing it.

josecelano commented on yarn.lock:
Hi @ivan9989-pkj, we are only using NPM for locking dependencies. You have to remove this `yarn.lock` file.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -2,7 +2,7 @@
 slug: how-to-setup-the-development-environment
 coverImage: /images/posts/development-environment.png
-date: 2023-07-11T12:29:04.295Z
+date: 2024-04-18T12:29:04.295Z
```
Hi @ivan9989-pkj, this is the publication date. You have to keep the original. [We want to add the update date in the future](https://github.com/torrust/torrust-website/issues/151).

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -56,16 +56,13 @@ You do not need to setup all services to contribute to the Torrust Index, but in
 Setting up the development environment requires to setup the three main services.
 
-First at all, we need to clone the repositories of the three services. We will clone them in a temporarily folder so you can follow this guide.
+First at all, we need to create the Tmp and torrust folders where we will store the repositories of our services.
```
Suggestion: 

"First, we need to create the folder (in this example, a temp folder) where we will store the repositories of our services."

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -86,28 +83,39 @@ sudo apt-get install libsqlite3-dev
 ## Set Up the Torrust Tracker
 
 At the time of writing, the Torrust Tracker requires:
 
-- rustc 1.72.0-nightly (839e9a6e1 2023-07-02)
+- rustc 1.77.1-nightly (839e9a6e1 2024-03-26)
```
HI @ivan9989-pkj I think we can change that because the tracker works with previous versions. Maybe we could say:

"This tutorial has been tested with the following rust version:"

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -86,28 +83,39 @@ sudo apt-get install libsqlite3-dev
+- rustc 1.77.1-nightly (839e9a6e1 2024-03-26)
 
-You can run the Tracker with the following commands:
+Now , we will create the tracker service and inside it we will create the folders where the databases will be located:
```
"Now, we will build the tracker and create the storage folders where persistent data like databases will be stored:"

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
````diff
@@ -86,28 +83,39 @@ sudo apt-get install libsqlite3-dev
 
-```s
-cd torrust-tracker/
-./bin/install.sh
-cargo run
+```bash
+  git clone https://github.com/torrust/torrust-tracker.git \
+  && cd torrust-tracker \
+  && cargo build --release \
+  && mkdir -p ./storage/tracker/lib/database \
+  && mkdir -p ./storage/tracker/lib/tls
 ```
 
 </CodeBlock>
````
Hi @ivan9989-pkj I would use this:

```
<CodeBlock lang="terminal">

`` `terminal
git clone https://github.com/torrust/torrust-tracker.git \\
  && cd torrust-tracker \\
  && cargo build --release \\
  && mkdir -p ./storage/tracker/lib/database \\
  && mkdir -p ./storage/tracker/lib/tls
`` `

</CodeBlock>
```

Otherwise the `\` at the end of the line is not shown.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -86,28 +83,39 @@ sudo apt-get install libsqlite3-dev
 </CodeBlock>
 
-The install script will generate:
+<!-- When executing the code it may not work for you , but it can be solved as easy as putting a \ at the end of each command. -->
```
This comment is not shown and it's essential to show the last line char `\`.

josecelano commented:
The output for the tracker has changed:

```output
    Finished `dev` profile [optimized + debuginfo] target(s) in 0.10s
     Running `target/debug/torrust-tracker`
Loading default configuration file: `./share/default/config/tracker.development.sqlite3.toml` ...
2024-04-22T16:24:23.241961292+01:00 [torrust_tracker::bootstrap::logging][INFO] logging initialized.
2024-04-22T16:24:23.242613018+01:00 [UDP TRACKER][INFO] Starting on: udp://0.0.0.0:6969
2024-04-22T16:24:23.242649758+01:00 [torrust_tracker::bootstrap::jobs][INFO] TLS not enabled
2024-04-22T16:24:23.242670038+01:00 [HTTP TRACKER][INFO] Starting on: http://0.0.0.0:7070
2024-04-22T16:24:23.242744307+01:00 [HTTP TRACKER][INFO] Started on: http://0.0.0.0:7070
2024-04-22T16:24:23.242753177+01:00 [torrust_tracker::bootstrap::jobs][INFO] TLS not enabled
2024-04-22T16:24:23.242834227+01:00 [API][INFO] Starting on http://127.0.0.1:1212
2024-04-22T16:24:23.242848277+01:00 [API][INFO] Started on http://127.0.0.1:1212
2024-04-22T16:24:23.242864747+01:00 [HEALTH CHECK API][INFO] Starting on: http://127.0.0.1:1313
2024-04-22T16:24:23.242897537+01:00 [HEALTH CHECK API][INFO] Started on: http://127.0.0.1:1313
```

You can update it.

josecelano commented:
Hi @ivan9989-pkj we should update this section too:

![image](https://github.com/torrust/torrust-website/assets/58816/9d86df80-64d0-4fac-b472-c9dc24a290e0)

By default, if you don't specify any `Config.toml` file, the application will use this:

```
Loading default configuration file: `./share/default/config/tracker.development.sqlite3.toml` ...
```

You can't change that file because it's a template included in the repo. If you want to set your custom configuration, you can either:

1.- Use a different path for the config file.
2.- Inyect the configuration with an env var.

### Custom config file

First, copy the template file to the storage folder:

```console
cp share/default/config/tracker.development.sqlite3.toml storage/tracker/etc/tracker.toml
```

Then, you can change any value and finally run the tracker with:

```console
TORRUST_TRACKER_PATH_CONFIG="./storage/tracker/etc/tracker.toml" cargo run
```

That would give you this outout:

```output
TORRUST_TRACKER_PATH_CONFIG="./storage/tracker/etc/tracker.toml" cargo run
    Finished `dev` profile [optimized + debuginfo] target(s) in 0.09s
     Running `target/debug/torrust-tracker`
Loading configuration file: `./storage/tracker/etc/tracker.toml` ...
2024-04-22T16:32:57.035457075+01:00 [torrust_tracker::bootstrap::logging][INFO] logging initialized.
2024-04-22T16:32:57.036048971+01:00 [UDP TRACKER][INFO] Starting on: udp://0.0.0.0:6969
2024-04-22T16:32:57.036079671+01:00 [torrust_tracker::bootstrap::jobs][INFO] TLS not enabled
2024-04-22T16:32:57.036117731+01:00 [HTTP TRACKER][INFO] Starting on: http://0.0.0.0:7070
2024-04-22T16:32:57.036209310+01:00 [HTTP TRACKER][INFO] Started on: http://0.0.0.0:7070
2024-04-22T16:32:57.036218550+01:00 [torrust_tracker::bootstrap::jobs][INFO] TLS not enabled
2024-04-22T16:32:57.036280190+01:00 [API][INFO] Starting on http://127.0.0.1:1212
2024-04-22T16:32:57.036283750+01:00 [API][INFO] Started on http://127.0.0.1:1212
2024-04-22T16:32:57.036301960+01:00 [HEALTH CHECK API][INFO] Starting on: http://127.0.0.1:1313
2024-04-22T16:32:57.036385919+01:00 [HEALTH CHECK API][INFO] Started on: http://127.0.0.1:1313
```

### Inyect the configuration with an env var

You can also inject the configuration with:

```console
TORRUST_TRACKER_CONFIG=`cat share/default/config/tracker.development.sqlite3.toml` cargo run
```

NOTICE: we load the whole file into the env var. This is not useful for development, but it's a different way to inject the configuration. It's used when running the tracker with docker.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -128,7 +136,7 @@ Loading configuration from config file ./config.toml
 </CodeBlock>
 
-By default, only the API is enabled, if you want to enable the HTTP or UDP trackers, you need to change the `enabled` value in the `./config.toml` file.
+<!-- By default, only the API is enabled, if you want to enable the HTTP or UDP trackers, you need to change the `enabled` value in the `./config.toml` file.
```
HI @ivan9989-pkj why is this comment hidden? We can show it. However, it's wrong. For the development default config now all services are enabled by default. See the tracker.development.sqlite3.toml file.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
````diff
@@ -207,12 +215,15 @@ cargo install sqlx-cli
+We will now put the torrust-backend repository, cloning it and saving it in the torrust folder:
 
 <CodeBlock lang="terminal">
 
 ```bash
-sudo apt-get install libsqlite3-dev
+  git clone https://github.com/torrust/torrust-index.git \
+  cd torrust-index \
+  && cargo build --release \
+  && mkdir -p ./storage/database
 ```
 
 </CodeBlock>
````
Same here. we need to show the `\` at the line end.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
```diff
@@ -222,17 +233,15 @@ You can run the Tracker Backend with the following commands:
 As you can see we are using the environment variable `TORRUST_IDX_BACK_CORS_PERMISSIVE` to enable a permissive CORS policy. The default port for the Backend is `3001` and for the web server serving the frontend application is `3000`. Since they are different ports, we need to tell the backend to allow request from a different port so that the frontend can make request to the API. To know more abo ...
 
 <Callout type="info">
-  The <code>install.sh</code> script will generate a SQLite database for the Tracker, but we are not going to use it. That one is only used if you want to run the tracker with docker. For example when you are working on features that only involves the Index and do not require changes in the Tracker. In this case, we are running the Tracker directly on its own folder. That could be useful if you w ...
+  The <code>install.sh</code> script will generate a SQLite database for the      Tracker, but we are not going to use it. That one is only used if you want to run the tracker with docker. For example when you are working on features that only involves the Index and do not require changes in the Tracker. In this case, we are running the Tracker directly on its own folder. That could be useful if  ...
 </Callout>
```
I think we can remove this callout. That script does not exist anymore.

josecelano commented on src/routes/(blog-article)/how-to-setup-the-development-environment/+page.md:
````diff
@@ -256,17 +265,28 @@ For more details about the Torrust Index Backend, check the [Index Backend docum
 
 The frontend is a [Nuxt](https://nuxt.com/) application.
 
+The last repository we need to upload is the torrust-index-gui repository . Remember to install Node.js and we need to be in the torrust folder to clone the repository.
+
+<CodeBlock lang="terminal">
+
+```bash
+  git clone https://github.com/torrust/torrust-index-gui.git
+
+  npm install
+```
+
+</CodeBlock>
````
The las repository we need to **download** ...

And you can change the command to:

```
git clone https://github.com/torrust/torrust-index-gui.git
cd torrust-index-gui.git
npm install
npm run dev
```

TO use one single command.

josecelano commented:
Hi @ivan9989-pkj we should also update screenshots:

![image](https://github.com/torrust/torrust-website/assets/58816/79e91612-4216-4c18-98a0-9a674c743fcd)

josecelano reviewed (changes requested):
Hi @ivan9989-pkj overall it is good but we need to fix all these things to make it as clear as possible for newcomers.

ivan9989-pkj committed:
How-to-setup-the-development-environment update

We changed everything requested by jose celano and upgraded to a more modern version of the tracker.

josecelano commented:
Hi, @ivan9989-pkj, I've added your changes and other fixes via a new [PR](https://github.com/torrust/torrust-website/pull/165).
